my/core/query.py

```diff
@@ -54,6 +54,15 @@ def locate_function(module_name: str, function_name: str) -> Callable[[], Iterab
         for (fname, func) in inspect.getmembers(mod, inspect.isfunction):
             if fname == function_name:
                 return func
+        try:
+            # incase the function is defined dynamically,
+            # like with a globals().setdefault(...) or a module-level __getattr__ function
+            func = getattr(mod, function_name)
+            if callable(func):
+                return func
+        except AttributeError:
```

Comment:
hm, wouldn't it be cleaner to `hasattr`, or pass `None` as default to `getattr` and check it later?

Reply:
Ah, true. I think the `getattr` with `None` works better here because of the additional check to see if its callable